tests/PerformanceTests/PerformanceTestService/EventSource/MemoryCacheEventSource.cs

```diff
@@ -54,32 +63,80 @@ public void IncrementReadCount()
             Interlocked.Increment(ref _readCount);
         }
 
+        public void IncrementReadMissCount()
+        {
+            Interlocked.Increment(ref _readMissCount);
+        }
+
         public void IncrementRemoveCount()
         {
             Interlocked.Increment(ref _removeCount);
             Interlocked.Decrement(ref _totalItemCount);
         }
+        
+        public void IncrementSize(int sizeInBytes)
+        {
+            Interlocked.Add(ref _cacheSizeInBytes, sizeInBytes);
+        }
+
+        public void DecrementSize(int sizeInBytes)
+        {
+            Interlocked.Add(ref _cacheSizeInBytes, sizeInBytes * -1);
+        }
+
+        public void AddReadDuration(double readDurationInMilliseconds)
+        {
+            _cacheReadDurationCounter.WriteMetric(readDurationInMilliseconds);
+        }
 
+        public void AddWriteDuration(double writeDurationInMilliseconds)
+        {
+            _cacheWriteDurationCounter.WriteMetric(writeDurationInMilliseconds);
+        }
+
+        /// <remarks>
+        /// This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
+        /// They aren't disabled afterwards
+        /// </remarks>
         protected override void OnEventCommand(EventCommandEventArgs command)
         {
             if (command.Command == EventCommand.Enable)
             {
-                _cacheItemCounter ??= new PollingCounter(CacheItemCounterName, this, () => _totalItemCount)
+                _cacheItemCounter ??= new PollingCounter(CacheItemCounterName, this, () => Volatile.Read(ref _totalItemCount))
```

Comment:
FYI there was a discussion whether `Volatile.Read` or `Interlocked.Read` should be used.
The [consensus](https://github.com/dotnet/docs/pull/20984#discussion_r502028017) was to use `Interlocked.Read` in this case.

Reply:
Awesome, thanks @gfoidl. This is very helpful.